src/Symfony/Component/Dsn/Tests/Factory/MemcachedFactoryTest.php

```diff
@@ -0,0 +1,154 @@
+    {
+        MemcachedFactory::create(array(), array($name => $value));
+    }
+
+    public function provideBadOptions()
+    {
+        return array(
+            array('foo', 'bar'),
+            array('hash', 'zyx'),
+            array('serializer', 'zyx'),
+            array('distribution', 'zyx'),
+        );
+    }
+
+    public function testDefaultOptions()
+    {
+        $client = MemcachedFactory::create(array());
+
+        $this->assertTrue($client->getOption(\Memcached::OPT_COMPRESSION));
+        $this->assertSame(1, $client->getOption(\Memcached::OPT_BINARY_PROTOCOL));
+        $this->assertSame(1, $client->getOption(\Memcached::OPT_LIBKETAMA_COMPATIBLE));
+    }
+
+    /**
+     * @dataProvider provideServersSetting
+     */
+    public function testServersSetting($dsn, $host, $port)
+    {
+        $client1 = MemcachedFactory::create($dsn);
+        $client2 = MemcachedFactory::create(array($dsn));
+        $expect = array(
+            'host' => $host,
+            'port' => $port,
+        );
+
+        $f = function ($s) { return array('host' => $s['host'], 'port' => $s['port']); };
+        $this->assertSame(array($expect), array_map($f, $client1->getServerList()));
+        $this->assertSame(array($expect), array_map($f, $client2->getServerList()));
+    }
+
+    public function provideServersSetting()
+    {
+        yield array(
+            'memcached://127.0.0.1/50',
+            '127.0.0.1',
+            11211,
+        );
+        yield array(
+            'memcached://localhost:11222?weight=25',
+            'localhost',
+            11222,
+        );
+        if (ini_get('memcached.use_sasl')) {
+            yield array(
+                'memcached://user:password@127.0.0.1?weight=50',
+                '127.0.0.1',
+                11211,
+            );
+        }
+        yield array(
+            'memcached:///var/run/memcached.sock?weight=25',
+            '/var/run/memcached.sock',
+            0,
+        );
+        yield array(
+            'memcached:///var/local/run/memcached.socket?weight=25',
+            '/var/local/run/memcached.socket',
+            0,
+        );
+        if (ini_get('memcached.use_sasl')) {
+            yield array(
+                'memcached://user:password@/var/local/run/memcached.socket?weight=25',
+                '/var/local/run/memcached.socket',
+                0,
+            );
+        }
+    }
+
+    /**
+     * @dataProvider provideDsnWithOptions
+     */
+    public function testDsnWithOptions($dsn, array $options, array $expectedOptions)
+    {
+        $client = MemcachedFactory::create($dsn, $options);
+
+        foreach ($expectedOptions as $option => $expect) {
+            $this->assertSame($expect, $client->getOption($option));
+        }
+    }
+
+    public function provideDsnWithOptions()
+    {
+        if (!class_exists('\Memcached')) {
+            self::markTestSkipped('Extension memcached required.');
```

Comment:
`Extension memcached required` -> `memcached PHP extension required.` ?

Reply:
removed => already handled by ` * @requires extension memcached` in the class's header